services/management/mongodb.go

```diff
@@ -72,6 +72,10 @@ func (s *MongoDBService) Add(ctx context.Context, req *managementpb.AddMongoDBRe
 
 		mongoDBOptions := models.MongoDBOptionsFromRequest(req)
 
+		if req.AddNode.NodeType == inventorypb.NodeType_REMOTE_NODE {
```

Comment:
Why do we check Node type, of all things, to determine if Agent should use pull mode?

Reply:
In case of remote we always set pull mode because of "Add Instance" from UI.